sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -91,5 +99,73 @@ internal static string GenerateUserAgentString(Assembly clientAssembly, string?
         /// The properly formatted UserAgent string based on this <see cref="TelemetryDetails"/> instance.
         /// </summary>
         public override string ToString() => _userAgent;
+
+        private static string EnsureProperParenthesisMatching(string userAgent)
+        {
+            var stack = new Stack<char>();
+            bool needsFix = false;
+            StringBuilder? fixedUserAgent = null;
+            for (int i = 0; i < userAgent.Length; i++)
+            {
+                switch (userAgent[i])
+                {
+                    case '(':
+                        stack.Push('(');
```

Comment:
We seem to only ever append open parens - why do we even need a stack? Could we just keep track of the open - close parens count?

Reply:
Yeah, this will just ensure that we have the correct number of matching or will remove closed parentheses that don't have a matching open. I don't think just the count is sufficient unless we only wanted to validate without fixing.

Reply:
But if we are only pushing one type of element, how I don't see why the stack is needed.

Reply:
oops - yes, you are correct. Fixed.